core/camel-core/src/test/java/org/apache/camel/converter/ConverterTest.java

```diff
@@ -168,38 +174,38 @@ public void testFileToString() throws Exception {
     }
 
     @Test
-    public void testPrimitiveBooleanConversion() throws Exception {
-        boolean value = converter.convertTo(boolean.class, null);
+    public void testPrimitiveBooleanConversion() {
+        boolean value = assertDoesNotThrow(() -> converter.convertTo(boolean.class, null));
```

Comment:
Brings more complexity to the test and reduces its readability for no real gain IMHO

Reply:
It is an expectation for this test that when receiving a primitive type, the conversion succeeds. By allowing the exception to leak, it could potentially be misinterpreted as some generic fail. Let's add a fail message here, and be clear about out expectations (same for the other ones).

Reply:
One more thing in favor of the `assertDoesNotThrow` with the fail message: we get more details on the CI job in case of failure. That prevents us from having to hunt down the logs on the CI.